cmd/clusterctl/client/config_test.go

```diff
@@ -289,6 +289,16 @@ func Test_clusterctlClient_templateOptionsToVariables(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "tolerates subdomains as cluster Name",
+			args: args{
+				options: GetClusterTemplateOptions{
+					ClusterName:     "foo.bar",
+					TargetNamespace: "baz",
+				},
+			},
+			wantErr: false,
```

Comment:
_very_ minor nit: can this test case be moved above to be grouped with all the success cases?